### PR TITLE
Migrate to klog from logrus

### PR DESCRIPTION
**Pull request by bormanp**
I will note that klog defaults to --logtostderr=true but in kne I have defaulted it to false as one reason to go to klog is to have real log files.

### CODE REVIEW

**Comment by bormanp** [comment]
I should point out this is a user facing change, users will notice.  Are there documents we should modify?

File: deploy/deploy.go [hunk 1]
```diff
@@ -510,7 +510,17 @@ func (k *KindSpec) setupGoogleArtifactRegistryAccess() error {
...
 	}
-	execer.SetStdout(log.StandardLogger().Out)
+	// It would be nice to something like:
+	//
+	//	errlog := logshim.New(func(args ...interface{}) {
+	//		log.InfoDepth(2, args...)
+	//	})
+	// So all the log lines don't just say they are coming from logshim, but
+	// it really doesn't help because the lines are coming from a goroutine
+	// that is using io.Copy.  So we just let the logs say logshim.
```
Comment (alexmasi) [inline deploy/deploy.go:520]: nit: maybe clean up this comment, not sure how much clarity it adds in its current state

File: deploy/deploy.go [hunk 2]
```diff
@@ -48,10 +49,9 @@ data:
 )
 
 var (
-	logOut        = log.StandardLogger().Out
+	logOut        = os.Stderr
```
Comment (alexmasi) [inline deploy/deploy.go:52]: so the logs originally go to stderr until line 523?

shouldn't this line setup the logshim?

**Comment by bormanp** [comment]
I have made the default to log to both files and stderr (otherwise it is silent).

**Review by alexmasi — APPROVED** [review]

**Pull request merged**